docs/npm.md

```diff
@@ -0,0 +1,458 @@
+      "version": "1.0.0",
+      "resolved": "https://github.com/cachito-testing/cachito-npm-without-deps/raw/tarball/cachito-npm-without-deps-1.0.0.tgz",
+      "integrity": "sha512-Q+cfkK1fnrNJqxiig/iVSZTe83OWLdxhuGa96k1IJJ5nkTxrhNyh6MUZ6YHKH8xitDgpIQSojuntctt2pB7+3g=="
+    },
+    "node_modules/csstype": {
+      "version": "3.1.2",
+      "resolved": "https://registry.npmjs.org/csstype/-/csstype-3.1.2.tgz",
+      "integrity": "sha512-I7K1Uu0MBPzaFKg4nI5Q7Vs2t+3gWWW648spaF+Rg7pI9ds18Ugn+lvg4SHczUdKlHI5LWBXyqfS8+DufyBsgQ=="
+    },
+    "node_modules/fecha": {
+      "version": "4.2.3",
+      "resolved": "file:fecha-4.2.3.tgz",
+      "integrity": "sha512-OP2IUU6HeYKJi3i0z4A19kHMQoLVs4Hc+DPqqxI2h/DPZHTm/vjsfC6P0b4jCMy14XizLBqvndQ+UilD7707Jw==",
+      "license": "MIT"
+    },
+    "node_modules/foo": {
+      "resolved": "foo",
+      "link": true
+    },
+    "node_modules/is-positive": {
+      "version": "3.1.0",
+      "resolved": "git+ssh://git@github.com/kevva/is-positive.git#97edff6f525f192a3f83cea1944765f769ae2678",
+      "license": "MIT",
+      "engines": {
+        "node": ">=0.10.0"
+      }
+    }
+  }
+}
+
+```
+</details>
+
+<details>
+<summary>foo/package.json (workspace)</summary>
+
+```jsonc
+{
+  "name": "foo",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "devDependencies": {},
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+      "is-positive": "github:kevva/is-positive"
+  }
+}
+```
+</details>
+
+
+## Using fetched dependencies
+
+See also [usage.md](usage.md) for a complete example of Cachi2 usage.
+
+Cachi2 downloads the npm dependencies as tar archives into the `deps/npm/` subpath of the output directory.
+
+1. Dependencies fetched from npm registries are placed directly to this directory (array-flatten in the following example).
+1. Dependencies downloaded from other HTTPS URL are placed to subdirectory `external-<tarball_name>` (cachi2-bar in the following example).
+1. Dependencies retrieved from Git repository are placed to `host, namespace, repo` subdirectories (cachi2-foo in the following example).
+
+```text
+cachi2-output/deps/npm
+├── array-flatten-1.1.1.tgz
+├── bitbucket.org
+│        └── foo-testing
+│             └── cachi2-foo
+│                       └── cachi2-foo-external-gitcommit-9e164b97043a2d91bbeb992f6cc68a3d1015086a.tgz
+├── body-parser-1.20.1.tgz
+├── bytes-3.1.2.tgz
+│   ...
+├── external-cachi2-bar
+│        └── cachi2-bar-external-sha512-43e71f90ad5YOLO.tgz
+│   ...
+```
+
+Cachi2 updates [required files](#required-files) for offline installation.
+
+Update root package.json file together with package.json files for each [workspace](https://docs.npmjs.com/cli/v9/using-npm/workspaces?v=true) with changes:
+* For git repositories and HTTPS URLs in dependencies update their value to an empty string
+
+Update the following in the package-lock.json file:
```

Comment:
I think it's important to emphasize that the updates are performed automatically by Cachi2's `inject-files` command, and that we're just describing here what the changes are. The common user probably won't go deep into this.

Reply:
True, even from my Demo was not obvious I was explaining functionality not actual usage. I updated it. Is it better?

Reply:
I'll close this comment and add three more comments about this section, in their respective lines.